### ISSUE TITLE
convert range to Miles depending on the country code 

### ISSUE THREAD

**Issue by lonebaggie** [epenet/hassRenaultZE#54]
I know its simple enough to convert km to miles  using a template sensor, but since the config-flow already knows the country code , it would be nice to have miles rather than Km 

P.s the new integration is so much better, no errors, much quicker to load , stunning work thank-you

**Comment by epenet** [comment]
What unit system are you using in your global Home Assistant configuration?

If you look at these two lines, we check the global configuration to see if you are set to use metric (Celsius, kilograms) or imperial (Fahrenheit, pounds)

Mileage sensor:
https://github.com/epenet/hassRenaultZE/blob/develop/custom_components/renault/sensor.py#L236

Range sensor:
https://github.com/epenet/hassRenaultZE/blob/develop/custom_components/renault/sensor.py#L258

If you feel that an "options flow" would be useful to override that, then I'm open to PRs

**Comment by lonebaggie** [comment]
Ah I see ,  We are very confused in the UK we use a metric system, except for miles , and the drinks in Pubs (pints)  . If I change the global setting to Imperial , I will lose all my metric temperatures and weights, only the British !

**Comment by oncleben31** [comment]
I'm convince the issue should be solved at Home Assistant core side. Using the user `unit system` HA settings is the best approach to define the unit used.

**Comment by epenet** [comment]
I've seen other integrations using an extra "options flow" (better than rely on the country-code in my opinion)

If you want to provide a PR for this I'll be happy to integrate...

If not, then at the moment I am working hard on approvals for the "onewire" integration (https://github.com/home-assistant/core/pulls 42041, 42037, 41826, 40943, 40786, 40774, 40680, 40550, 40530, 40528, 39780, 39321 and extra ones pending on my machine). Once all those are out of the way then I can look at this again.

**Comment by lonebaggie** [comment]
It appears to be more of a HA issue to allow a  more granular use of the unit system  .  The Americans must  have the same problem with imperial gallons.  
Its nothing important , A template sensor will work fine , it can wait

**Comment by epenet** [comment]
There is an open discussion about it on https://github.com/home-assistant/architecture/issues/10

Maybe this is needed after all

**Comment by epenet** [comment]
@alexanderfitu and all,
I have created a PR #102 to implement this.

If you want to try it you can copy from this branch: https://github.com/hacf-fr/hassRenaultZE/tree/distances-in-miles

**Comment by lonebaggie** [comment]
Works fine . Thank you

**Comment by alexanderfitu** [comment]
I am running the PR 102, but cannot see where the option is to change distances to miles?

**Comment by epenet** [comment]
`Configuration` > `Integrations` > `Renault`
You should have two links => `Rename` and `Options`
Choose `Options`

**Comment by alexanderfitu** [comment]
I thought so, but I only get these two options:

![image](https://user-images.githubusercontent.com/3074998/103585990-15c5fa80-4edc-11eb-842a-dfd1db039dd2.png)

**Comment by epenet** [comment]
You have an issue with the translation file. Maybe simply a cache issue?

- `300` is the interval between API calls (in seconds)
- the `checkbox` is to display in miles

**Comment by alexanderfitu** [comment]
Possibly, I have tried in incognito tab, but labels are missing. 

In any case, ticking the box does change to miles! Many thanks.

**Comment by epenet** [comment]
Strange - the text works for me in French

**Comment by lonebaggie** [comment]
![image](https://user-images.githubusercontent.com/23200848/103647879-e226b700-4f53-11eb-9ccd-453feeb44325.png)

OK in English

**Comment by alexanderfitu** [comment]
My unit system is metric in HA and I chose en-GB when setting up the plugin. My language is also English.

**Comment by lonebaggie** [comment]
Upload new version . All good :) 

![image](https://user-images.githubusercontent.com/23200848/103683496-2df05500-4f82-11eb-9743-8402bc594c4a.png)

**Comment by epenet** [comment]
Thanks @lonebaggie I will merge the PR.
